vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,599 @@
+of the third dimension if there is a time dimension).
+
+```{r, VertAtt, results = 'hide'}
+vertical_attributes <- list(
+  name = 'depth',
+  units = 'cm',
+  positive = 'down', # depth is down, height is up
+  vals = depths_max
+)
+```
+
+### CRS Attributes
+
+The CRS attributes describe the coordinate reference system (CRS).
+
+The only required field in the CRS attribute list is the 'proj' field. The proj
+field is the the CRS is proj4string from the PROJ.4 library.
+
+Additionally, we strongly recommend defining the CRS in the crs_wkt format, which
+is preferred by [netCDF convention] (http://cfconventions.org/cf-conventions/cf-conventions.html#use-of-the-crs-well-known-text-format).
+
+The *wkt* function in the raster package can be used to generate wkts.
+
+```{r, CRSAtt, results = 'hide'}
+
+crs_attributes <- list(
+ proj = '+init=epsg:4326',
+  grid_mapping_name = "latitude_longitude",
+  longitude_of_prime_meridian = 0.0,
+  semi_major_axis = 6378137.0,
+  inverse_flattening = 298.257223563
+)
+
+locationsSP <- sp::SpatialPoints(locations)
+sp::proj4string(locationsSP) <- sp::CRS(crs_attributes$proj)
+WKTex <- raster::wkt(locationsSP)
+
+crs_attributes <- c(crs_attributes, crs_wkt = WKTex)
+
+```
+
+### Global Attributes
+
+A named list describing the global attributes of the netCDF. This is where you describe
+things like how the data was made, where the data was made, who the data was made by,
+what climate models were used to create the data, etc.
+
+The *title*, *institution*, *description*, *source_id*, *realm*, *product*, and
+*contact* fields are required.
+
+```{r, GlobalAtt, results = 'hide'}
+global_attributes <- list(
+    title = "Project Name Annual Wet Degree Days",
+    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+    description = 'methods on how this data was made',
+    source = paste(
+      "SOILWAT2 (v4.2.0);",
+      "rSOILWAT2 (v2.3.2);",
+      "rSFSW2 (v3.1.2)"
+    ),
+    source_id = "SOILWAT2",
+    realm = "land",
+    parent_mip_era = "CMIP5",
+    parent_experiment_id = "RCP45",
+    parent_source = "CanESM2",
+    parent_variant_label = "r1i1p1f1",
+    product = "model-output",
+    projection = 'Geographic',
+    grid = 'WGS84',
+    grid_label = "gn",
+    nominal_resolution = "10 km", # \code{\link{populate_netcdf_from_array}}
+    further_info_url = "https://github.com/DrylandEcology/",
+    contact = "you@email.com"
+    )
+```
+
+#### Nominal resolution
+
+If the data is calculated and reported on a grid, [CMIP6 convention](https://docs.google.com/document/d/1h0r8RZr_f3-8egBMMh7aqLwy3snpD6_MrDz1q8n5XUk/edit)
+calls for the field *nominal resolution* to be contained in the global attributes.
+The function \code{calculate_nominal_resolution} will calculate the nominal resolution
+of your gridded data for you. However, you need to provide a raster of the grid,
+the lat/long locations, and the area that each cell in the grid represents.
```

Comment:
Here and throughout, I think we need to be careful about whether we refer to data in lat/long coordinate pairs or long/lat coordinates.

See [R spatial follows GDAL and PROJ development]{https://www.r-spatial.org/r/2020/03/17/wkt.html}:
- PROJ respects authorities (such as EPSG) for determining whether coordinate pairs refer to longitude-latitude (such as 3857), or latitude-longitude (such as 4326)
- GDAL offers the ability to handle coordinate pairs authority-compliant (lat-long for 4326), or “traditional” GIS-compliant (long-lat for 4326)
- `sf::st_axis_order`
- Workflows using sp/rgdal should expect “GIS style” axis order to be preserved

Reply:
Hmm - Could you point to where you want this changed? Because this file is now outdated, I am no longer able to find where you refer to.